src/main.go

```diff
@@ -0,0 +1,241 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+const (
+	githubContextEnv = "GITHUB_CONTEXT"
+	jobContextEnv    = "JOB_CONTEXT"
+)
+
+type WorkflowNotificationCommand struct {
+	cli.BaseCommand
+	flagWebhookUrl string
+}
+
+func (c *WorkflowNotificationCommand) Desc() string {
+	return "Send a message to a Chat space"
+}
+
+func (c *WorkflowNotificationCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  The chat command sends messages to Chat spaces.
+`
+}
+
+func (c *WorkflowNotificationCommand) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet()
+
+	f := set.NewSection("Chat space options")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "webhook_url",
+		Aliases: []string{"w"},
+		Example: "https://chat.goog...",
+		Default: "",
+		Target:  &c.flagWebhookUrl,
+		Usage:   `Webhook URL from google chat`,
+	})
+
+	return set
+}
+
+func (c *WorkflowNotificationCommand) Run(ctx context.Context, args []string) error {
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	args = f.Args()
+	if len(args) != 0 {
+		return fmt.Errorf("expected 0 arguments, got %q", args)
+	}
+
+	ghJsonStr := os.Getenv(githubContextEnv)
+	if ghJsonStr == "" {
+		return fmt.Errorf("environment var %s not set", githubContextEnv)
+	}
+	jobJsonStr := os.Getenv(jobContextEnv)
+	if jobJsonStr == "" {
+		return fmt.Errorf("environment var %s not set", jobContextEnv)
+	}
+
+	ghJson := map[string]any{}
+	jobJson := map[string]any{}
+	if err := json.Unmarshal([]byte(ghJsonStr), &ghJson); err != nil {
+		return fmt.Errorf("failed unmarshaling %s: %w", githubContextEnv, err)
+	}
+	if err := json.Unmarshal([]byte(jobJsonStr), &jobJson); err != nil {
+		return fmt.Errorf("failed unmarshaling %s: %w", jobContextEnv, err)
+	}
+
+	b, err := messageBody(ghJson, jobJson)
+	if err != nil {
+		return fmt.Errorf("failed to generate message body: %w", err)
+	}
+
+	url := c.flagWebhookUrl
+	fmt.Println("url: ", url)
+
+	request, err := http.NewRequest("POST", url, bytes.NewBuffer(b))
+	if err != nil {
+		return fmt.Errorf("creating http request failed: %w", err)
+	}
+	resp, err := http.DefaultClient.Do(request)
+	if err != nil {
+		return fmt.Errorf("sending http request failed: %w", err)
+	}
+	fmt.Println("resp: ", resp)
+	defer resp.Body.Close()
+
+	return nil
+}
+
+func main() {
+	if err := realMain(); err != nil {
```

Comment:
The main function should create a signal context so that the program exits on ctrl+c:

```go
ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
defer cancel()
```

See our other projects for an example.

Reply:
Updated.